support-frontend/assets/helpers/subscriptionsForms/formActions.js

```diff
@@ -48,7 +49,16 @@ export type Action =
 
 // ----- Action Creators ----- //
 
-const setStage = (stage: Stage): Action => ({ type: 'SET_STAGE', stage });
+const setStage = (
+  stage: Stage,
+  product: SubscriptionProduct,
+  paymentMethod: ?PaymentMethod,
+): Action => {
+  if (stage === 'thankyou' || stage === 'thankyou-pending') {
+    trackThankYouPageLoaded(product, paymentMethod);
```

Comment:
I'm a redux novice, but it's not really an action "creator" if it has the side effect of doing tracking?  could the tracking be when the action is fired not when it's created or is this the standard pattern?

Reply:
It is the pattern we generally use in support-frontend, there is a discussion about various approaches here: https://goshakkk.name/redux-side-effect-approaches/